server/service/src/invoice/outbound_return/delete/mod.rs

```diff
@@ -0,0 +1,211 @@
+        fn wrong_store() -> InvoiceRow {
+            InvoiceRow {
+                id: "wrong_store".to_string(),
+                store_id: mock_store_a().id,
+                r#type: InvoiceRowType::OutboundReturn,
+                name_link_id: mock_name_store_a().id,
+                ..Default::default()
+            }
+        }
+        fn verified() -> InvoiceRow {
+            InvoiceRow {
+                id: "verified".to_string(),
+                store_id: mock_store_b().id,
+                r#type: InvoiceRowType::OutboundReturn,
+                name_link_id: mock_name_store_b().id,
+                status: InvoiceRowStatus::Verified,
+                ..Default::default()
+            }
+        }
+
+        let (_, _, connection_manager, _) = setup_all_with_data(
+            "delete_outbound_return_errors",
+            MockDataInserts::all(),
+            MockData {
+                invoices: vec![wrong_store(), verified()],
+                ..Default::default()
+            },
+        )
+        .await;
+
+        let service_provider = ServiceProvider::new(connection_manager, "app_data");
+        let context = service_provider
+            .context(mock_store_b().id, "".to_string())
+            .unwrap();
+        let service = service_provider.invoice_service;
+
+        // InvoiceDoesNotExist
+        assert_eq!(
+            service.delete_outbound_returns(&context, vec!["invalid".to_string()]),
+            Err(ServiceError::InvoiceDoesNotExist)
+        );
+
+        //NotAnOutboundReturn
+        assert_eq!(
+            service.delete_outbound_returns(&context, vec![mock_outbound_shipment_a().id]),
+            Err(ServiceError::NotAnOutboundReturn)
+        );
+
+        //NotThisStoreInvoice
+        assert_eq!(
+            service.delete_outbound_returns(&context, vec![wrong_store().id]),
+            Err(ServiceError::NotThisStoreInvoice)
+        );
+
+        //CannotEditFinalised
+        assert_eq!(
+            service.delete_outbound_returns(&context, vec![verified().id]),
+            Err(ServiceError::CannotEditFinalised)
+        );
+    }
+
+    #[actix_rt::test]
+    async fn delete_outbound_return_success() {
+        let (_, connection, connection_manager, _) =
+            setup_all("delete_outbound_return_success", MockDataInserts::all()).await;
+
+        let service_provider = ServiceProvider::new(connection_manager, "app_data");
+        let context = service_provider
+            .context(mock_store_b().id, "".to_string())
+            .unwrap();
+        let service = service_provider.invoice_service;
+
+        service
+            .delete_outbound_returns(&context, vec![mock_outbound_return_a().id])
+            .unwrap();
+
+        //test entry has been deleted
+        assert_eq!(
+            InvoiceRowRepository::new(&connection)
+                .find_one_by_id_option(&mock_outbound_return_a().id)
+                .unwrap(),
+            None
```

Comment:
Slightly tempted to suggest testing that stocklines are updated if picked as per `delete_stock_out_line`
It should be covered in the other test cases, but would be nice to catch if someone tries to `improve` this function in the future and try to say call the repository layer directly.
Probably not likely but was trying to understand what the expected behavour is here.